### PR TITLE
Initial checkin for openshift azure provider, cleans up components no…

### PR DESCRIPTION
**Pull request by awesomenix**
…t required, decouples cluster and adds support for UserData

**What this PR does / why we need it**:

1. Removes unnecessary code
2. Decouples Cluster object from machine, using openshift machine api
3. Uses UserData as a bootstrapping script

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -38,15 +10,48 @@ required = [
...
 
-[[constraint]]
-  name = "github.com/golang/mock"
-  version = "1.2.0"
+[[override]]
+  name = "k8s.io/kube-aggregator"
+  version = "kubernetes-1.12.5"
+
+[[override]]
+  name = "github.com/openshift/cluster-api"
```
Comment (ingvagabund) [inline Gopkg.toml]: We have just bumped to the latest cluster-api and k8s 1.13. Can you sync with that as well?

https://github.com/openshift/cluster-api-provider-aws/blob/master/Gopkg.toml#L31-L61

**Comment by awesomenix** [comment]
Thanks @enxebre for the review, i was holding off on addressing the changes to make sure its in a working state (creates a working cluster with this PR), that said, agree that we need to have minimal divergence from Upstream, Will send out mini PR's addressing the following

- Allowing userdata/remove hard dep on kubeadm
- Remove cluster obj hard dependency
- swap to openshit/cluster-api

this PR would eventually be abandoned in favour of other PR's (but signifies end result)

**Pull request closed without merging**